File: CharSort/Entity/InCharOutObjCalculator.cs
using System.Collections.Generic;
using CharSort.Interface;
using CharSort.Facade;
using System.Linq;

namespace CharSort.Entity
{
    internal class InCharOutObjCalculator : IInCharOutObjCalculator
    {
        public IList<InChar> CalculateInCharOutObjects(IList<InChar> inChars)
        {
            var csModel = CSFacade.GetCSModel();
            var retValue = new List<InChar>();
            var letters = inChars.Where(s => (s.InCharAscii >= csModel.startASCIILetters && s.InCharAscii <= csModel.endASCIILetters)).ToList();
            var numbers = inChars.Where(s =>( s.InCharAscii >= csModel.startASCIINumbers && s.InCharAscii <= csModel.endASCIINumbers)).ToList();
            if (letters.Count == numbers.Count)
            {
                for (int i = 0; i < letters.Count; i++)
                {
                    retValue.Add(letters[i]);
                    retValue.Add(numbers[i]);
                }
                return retValue;
            }
            foreach (var item in letters)
            {
                retValue.Add(item);
            }
            foreach (var item in numbers)
            {
                retValue.Add(item);
            }

            return retValue;
        }
    }
}

File: JAVA/CharSort/Entity/InCharOutObjCalculator.java
package CharSort.Entity;

import CharSort.Interface.*;
import CharSort.Facade.*;
import CharSort.*;
import java.util.*;

public class InCharOutObjCalculator implements IInCharOutObjCalculator
{
	public final List<InChar> CalculateInCharOutObjects(List<InChar> inChars)
	{
		CharSort.model.ICSModel csModel = CSFacade.GetCSModel();
		ArrayList<InChar> retValue = new ArrayList<InChar>();
		ArrayList<Object> letters = inChars.stream().filter(s -> (s.InCharAscii >= csModel.getstartASCIILetters() && s.InCharAscii <= csModel.getendASCIILetters())).collect(Collectors.toList());
		ArrayList<Object> numbers = inChars.stream().filter(s -> (s.InCharAscii >= csModel.getstartASCIINumbers() && s.InCharAscii <= csModel.getendASCIINumbers())).collect(Collectors.toList());
		if (letters.size() == numbers.size())
		{
			for (int i = 0; i < letters.size(); i++)
			{
				retValue.add(letters.get(i));
				retValue.add(numbers.get(i));
			}
			return retValue;
		}
		for (Object item : letters)
		{
			retValue.add(item);
		}
		for (Object item : numbers)
		{
			retValue.add(item);
		}

		return retValue;
	}
}